pkg/controller/registry/resolver/steps.go

```diff
@@ -230,3 +284,39 @@ func NewServiceAccountStepResources(csv *v1alpha1.ClusterServiceVersion, catalog
 	}
 	return rbacSteps, nil
 }
+
+type optionalManifests struct {
+	Files []string `json:"manifests"`
+}
+
+// isOptional returns a func giving whether a resource is in the list of optional manifests
+// or false if any issue occurs.
+func isOptional(properties []*api.Property) func(key string) bool {
```

Comment:
This func is doing operation that is not matching to what is described in the EP. In the EP, it is file-name-based identifier but for this implementation, this is a domain-based identifier. If this implementation is the intended choice, then the EP should be updated so that reviewers can look back to EP for references. Otherwise, this is very confusing.

Reply:
Looking back at a previous comment, it seems the domain-based identifier is a change that you intended to do. I think an update PR for the EP is in order.

Reply:
Hi @dinhxuanvu yes this is a change. It was requested by @joelanford during review. I will create an update PR for the EP but wanted to avoid too many round trips. We have already had a fair amount here.